contracts/modules/DecentHatsCreationModule.sol

```diff
@@ -62,49 +62,41 @@ contract DecentHatsCreationModule is DecentHatsUtils {
      * We also make use of `KeyValuePairs` to associate the topHatId with the Safe.
      */
     function createAndDeclareTree(CreateTreeParams calldata params) external {
-        IHats hatsProtocol = params.hatsProtocol;
-        address hatsAccountImplementation = params.hatsAccountImplementation;
-        IERC6551Registry erc6551Registry = params.erc6551Registry;
-
         // Create Top Hat
         (uint256 topHatId, address topHatAccount) = _processTopHat(
-            hatsProtocol,
-            erc6551Registry,
-            hatsAccountImplementation,
+            params.hatsProtocol,
+            params.erc6551Registry,
+            params.hatsAccountImplementation,
             params.keyValuePairs,
             params.topHat
         );
 
         // Create Admin Hat
         uint256 adminHatId = _processAdminHat(
-            hatsProtocol,
-            erc6551Registry,
-            hatsAccountImplementation,
+            params.hatsProtocol,
+            params.erc6551Registry,
+            params.hatsAccountImplementation,
             topHatId,
             topHatAccount,
             params.moduleProxyFactory,
-            params.decentAutonomousAdminMasterCopy,
+            params.decentAutonomousAdminImplementation,
             params.adminHat
         );
 
-        for (uint256 i = 0; i < params.hats.length; ) {
-            HatParams memory hat = params.hats[i];
-            _processHat(
-                hatsProtocol,
-                erc6551Registry,
-                hatsAccountImplementation,
-                topHatId,
-                topHatAccount,
-                params.hatsModuleFactory,
-                params.hatsElectionsEligibilityImplementation,
-                adminHatId,
-                hat
-            );
-
-            unchecked {
-                ++i;
-            }
-        }
+        // Create Role Hats
+        CreateRoleHatsParams memory roleHatParams = CreateRoleHatsParams({
```

Comment:
Why not inline this in the call to `_processRoleHats?`

Reply:
Good call. Updated